File: flumeModel/src/main/java/com/sjgo/TypeInterceptor.java
package com.sjgo;

import org.apache.flume.Context;
import org.apache.flume.Event;
import org.apache.flume.interceptor.Interceptor;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

/**
 * 自定义拦截器
 */
public class TypeInterceptor implements Interceptor{

    private List<Event> events;

    //初始化
    public void initialize() {
        events = new ArrayList<Event>();
    }

    //拦截处理单个事件
    public Event intercept(Event event) {
        Map<String,String> headers = event.getHeaders();
        String body = new String(event.getBody());
        if (body.contains("hello")){
            headers.put("event","hello");
        }else {
            headers.put("event","no");
        }
        return event;
    }

    //处理批量事件
    public List<Event> intercept(List<Event> list) {

        events.clear();
        for (Event event : list) {
            events.add(intercept(event));
        }

        return events;
    }

    //关闭
    public void close() {

    }

    /**
     * 反射类构造拦截器
     */
    public static class Bulider implements Interceptor.Builder{

        public Interceptor build() {

            return new TypeInterceptor();
        }

        public void configure(Context context) {

        }
    }
}

File: flumeModel/src/test/scala/com/sjgo/TypeInterceptor.scala
package com.sjgo

import java.util

import org.apache.flume.{Context, Event}
import org.apache.flume.interceptor.Interceptor

class TypeInterceptor extends Interceptor{

  var eventLists : util.ArrayList [Event]  = _
  // 初始化
  override def initialize(): Unit = {
    eventLists = new util.ArrayList[Event]()
  }

  /**
   *
   * @param event
   * @return Event
   * 处理单个事件
   */
  override def intercept(event: Event): Event = {
    // 获取事件头
    val headers = event.getHeaders()

    // 获取事件body内容
    val body = event.getBody().toString()

    // 根据body中事件名添加头信息
    if (body.contains("hello")){
      headers.put("event","hello")
    }orElse{
      headers.put("event","no")
    }
    event


  }

  // 批量处理事件环节
  override def intercept(list: util.List[Event]): util.List[Event] = {

    eventLists.clear()

    import scala.collection.JavaConversions._
    for(event <- list){
      eventLists.add(intercept(event))
    }

    eventLists
  }

  // 关闭
  override def close(): Unit = ???

  object Bulider extends  Interceptor.Builder{
    override def build(): Interceptor = new TypeInterceptor()

    override def configure(context: Context): Unit = ???
  }
}
